pages/api/stripe-webhook/index.ts

```diff
@@ -118,27 +119,58 @@ async function checkoutSessionComplete(userToUpgrade: User, properties: Stripe.C
   return error;
 }
 
+async function splitPaymentIntent(paymentIntentId: string) {
+  // https://stripe.com/docs/expand/use-cases#stripe-fee-for-payment
+  const paymentIntent = await stripe.paymentIntents.retrieve(
+    paymentIntentId,
+    {
+      expand: ['latest_charge.balance_transaction'],
+    }
+  );
+
+  const charge = paymentIntent.latest_charge as Stripe.Charge | null | undefined;
```

Comment:
Could there be a race condition here? I see we pass the payment Id but it is kind of weird it is called latest charge

Reply:
I agree the latest_charge name is a bit strange... I was following the example here to get the Stripe fee: https://stripe.com/docs/expand/use-cases#stripe-fee-for-payment

Maybe there are scenarios where you could pay off a payment intent with multiple charges, but I'm assuming that shouldn't happen for our subscriptions...